### DISCUSSION TITLE
Grant-PnPAzureADAppSitePermission fails with "Microsoft.Online.DirectoryServices.UniqueKeyPropertyException"

### DISCUSSION THREAD

**Discussion by LijuV-MSFT** [pnp/powershell#1267] [Q&A]
I am running into this issue where the Grant-PnPAzureADAppSitePermission cmdlet does not appear to work. 
Here are my commands:

```
PS C:\Temp\> Grant-PnPAzureADAppSitePermission -AppId $appID -DisplayName $appDisplayName -Permissions Write -Site $SPOSiteUrl

Id    : aTowaS50fG1zLnNwLmV4dHw5YzFhNTA3Zi04YmFhLTQ5MjAtOTA5MC02OWEwMDZjNjc2YjVAOTMxOTVlOTgtMjdlMi00MWU2LWFmMzUtYzNmMDc0ZjZmMGJi
Roles : {write}
Apps  : {SPOforPBIApp, 9c1a507f-8baa-4920-9090-69a006c676b5}


PS C:\Temp\> Get-PnPAzureADAppSitePermission -AppIdentity $appID

Id    : aTowaS50fG1zLnNwLmV4dHw5YzFhNTA3Zi04YmFhLTQ5MjAtOTA5MC02OWEwMDZjNjc2YjVAOTMxOTVlOTgtMjdlMi00MWU2LWFmMzUtYzNmMDc0ZjZmMGJi
Roles : 
Apps  : {SPOforPBIApp, 9c1a507f-8baa-4920-9090-69a006c676b5}
```

As you can tell, there are no visible errors in the output of the command.
It worked once in the past, but all my attempts at running these commands again appear to fail.
Azure AD Audit Logs however show a failed UserManagement event:

```
**Activity**
Date: 10/17/2021, 5:20 PM
Activity Type: Add app role assignment grant to user
Correlation ID: f654d75a-2536-4450-b6e5-8431c4bd71e8
Category: UserManagement
Status: failure
Status reason: Microsoft.Online.DirectoryServices.UniqueKeyPropertyException

**Initiated by (actor)**
Type: User
Display Name: 
Object ID: (# Object ID of SPOAdmin)
IP address: xxxxxx
User Principal Name: SPOAdmin@xxxxxx.xxx

**Additional Details**
User-Agent: EvoSTS
AppId: 31359c7f-bd7e-475c-86db-fdb8c937548e (# Application ID of PnP Management Shell)

**Target**
Type: ServicePrincipal
Id: 6816a33d-f636-46bc-9245-5d34bfa79b88 (# Object ID of PnP Management Shell)
Display Name: PnP Management Shell
User Principal Name:
```

The user SPOAdmin is a global admin in the tenant and the primary admin of the SharePoint site.
The only thing I can think of is that I had deleted the PnP Management Shell after my first (successful attempt).

Can anyone shed some light into what's happening here?

**Comment by rpodric** [comment]
Did you ever figure it out? I'm also seeing that, albeit with a different app:
https://documentcloud.adobe.com/o365pdf/start.html

Tried in two tenants and over a period of a week.

**Reply by LijuV-MSFT** [reply]
No, never did; this was something I was testing with and so did not follow up.
